modules/layers/src/line-layer/line-layer.ts

```diff
@@ -182,10 +182,11 @@ export default class LineLayer<DataT = any, ExtraProps extends {} = {}> extends
     }
   }
 
-  draw({uniforms}): void {
+  draw({parameters, uniforms}): void {
```

Comment:
With the current caching mechanism, changing parameters will cause a pipeline to rebuild. Because each layer has a unique `polygonOffset` to battle z-fighting, calling `model.setParameters` will practically disable all program reuse. @ibgreen @donmccurdy

Reply:
Hmm... @ibgreen I assume this is the purpose of the new factory destroy policy 'never' setting? Let's test that and consider a future 'lazy' destroy policy (or similar) if the performance benefits are worthwhile?

Reply:
No the problem is not cache getting removed, it's that the hash is overly aggressive.

Reply:
Does the hash even need to include the parameters? Could we have a whitelist or exclude list?

Reply:
Discussing with @ibgreen in a separate channel - basically WebGPU render pipeline does not allow dynamically setting parameters so the unified API does not accept it.

Reply:
Based on ...

https://github.com/visgl/luma.gl/blob/c5c15fbb9ab2fcaad998d030eac9e6312dd70aad/modules/engine/src/lib/pipeline-factory.ts#L111-L115

... it might be optional for WebGL, but will require a new pipeline for WebGPU. My hope with the suggestion of a 'lazy' destroy policy would be that if we have N layers and 1 pipeline, we can at least keep N pipelines in the cache rather than creating and destroying pipelines N times per frame.

Reply:
I wonder if it would be possible to combat the z-fighting in the vertex shader via a uniform, something like `gl_Position.z += 0.0001 * layerIndex`

Reply:
Yes... we also have instance ID now with WebGL2.

The z fighting work around has long been due for an update. We definitely need to make some changes while working on WebGPU support.